### ISSUE TITLE
Copyright year

### ISSUE THREAD

**Issue by fpj** [pravega/pravega#3508]
**Problem description**
We have been using the year 2017 in our source file copyright header according to the legal requirements of Dell as of now all primary developers of the project have been Dell employees. 

This issue is to track the following changes that we need to implement:

- The year of the header is supposed to represent the year of first publication, which is currently incorrect for files created in 2018 and 2019. Consequently, we either change that or make a bulk delete of the year in the source header of all files. I recommend we do the latter and our legal supports that option as long as we do the other next step.
- Like with Apache projects, the year of copyright is specified in a NOTICE file, which we update every year. This is a much better option because we only need to update the year in one place. The NOTICE file is suppose to contain the following text:

```
Copyright (c) 2017-XXXX Dell Inc., or its subsidiaries. All Rights Reserved.
```

**Problem location**
Copyright.

**Suggestions for an improvement**

- [x] Delete the year in the license header across all files (0.5)
- [x] Delete the year in the license header across all files (0.6)
- [ ] Delete the year in the license header across all files (master)
- [x] Add a NOTICE file with the copyright statement

**Comment by RaulGracia** [comment]
Closed via #3559 and #3864.

**Comment by fpj** [comment]
This is not done, I have partially done for the `0.5` branch, but not for `master`.
